src/stores/zapPendingOperations/ZapPendingOperationsStore.ts

```diff
@@ -0,0 +1,199 @@
+  if (
+    typeof obj === 'object' &&
+    obj !== null &&
+    'value' in obj &&
+    'type' in obj
+  ) {
+    const typedValue = obj as { value: any; type: string };
+
+    if (typedValue.type === 'bigint') {
+      return BigInt(typedValue.value) as T;
+    } else if (typedValue.type === 'array') {
+      return typedValue.value.map(deserializeWithTypes) as T;
+    } else {
+      // For primitive types, just return the value
+      return typedValue.value as T;
+    }
+  }
+
+  // If it's not a typed object, it might be a plain object (shouldn't happen with new structure)
+  if (typeof obj === 'object' && obj !== null) {
+    const result: Record<string, unknown> = {};
+
+    for (const [key, value] of Object.entries(obj)) {
+      result[key] = deserializeWithTypes(value);
+    }
+
+    return result as T;
+  }
+
+  return obj as T;
+};
+
+interface PendingOperationsState<T extends WalletMethods> {
+  pendingOperations: Record<string, PendingOperationData<T>>;
+  promiseResolvers: Map<string, PromiseResolver<T>>;
+
+  // Actions
+  addPendingOperation: <K extends WalletMethods>(
+    id: string,
+    operationName: K,
+    args: WalletMethodArgsType<K>,
+    resolve: PromiseResolver<K>['resolve'],
+    reject: PromiseResolver<K>['reject'],
+  ) => void;
+
+  removePendingOperation: (id: string) => void;
+
+  getPromiseResolversForOperation: (
+    id: string,
+  ) => PromiseResolver<T> | undefined;
+
+  clearAll: () => void;
+}
+
+export const useZapPendingOperationsStore = createWithEqualityFn<
+  PendingOperationsState<WalletMethods>
+>()(
+  persist(
+    (set, get) => ({
+      pendingOperations: {},
+      promiseResolvers: new Map(),
+
+      addPendingOperation: (id, operationName, args, resolve, reject) => {
+        set((state) => ({
+          pendingOperations: {
+            ...state.pendingOperations,
+            [id]: {
+              operationName,
+              args,
+              timestamp: Date.now(),
+              id,
+            },
+          },
+        }));
+
+        // Store promise resolvers in memory (not persisted)
+        get().promiseResolvers.set(id, {
+          resolve: resolve as (
+            value: WalletMethodReturnType<WalletMethods>,
+          ) => void,
+          reject,
+        });
+      },
+
+      removePendingOperation: (id) => {
+        set((state) => {
+          const newPendingOperations = { ...state.pendingOperations };
+          delete newPendingOperations[id];
+          return { pendingOperations: newPendingOperations };
+        });
+
+        // Remove promise resolvers from memory
+        get().promiseResolvers.delete(id);
+      },
+
+      getPromiseResolversForOperation: (id) => {
+        return get().promiseResolvers.get(id);
+      },
+
+      clearAll: () => {
+        set({ pendingOperations: {} });
+        get().promiseResolvers.clear();
+      },
+    }),
+    {
+      name: 'zap-pending-operations-storage',
+      version: 1,
```

Comment:
Do you know what type of change would break that store and require us to bump this value and forget about pending operation?